### PR TITLE
serializing control properties in alphabetical order

### PR DESCRIPTION
**Pull request by mizrael**
serializing control properties in alphabetical order

## Validation

Unit tests

### CODE REVIEW

**Review by tehcrashxor — APPROVED** [review]

File: src/Microsoft.PowerPlatform.PowerApps.Persistence/Models/Text.cs [hunk 1]
```diff
@@ -5,11 +5,11 @@
 
 namespace Microsoft.PowerPlatform.PowerApps.Persistence.Models;
 
-[FirstClass(nameof(Label))]
-public record Label : Control
```
Comment (petrochuk) [inline src/Microsoft.PowerPlatform.PowerApps.Persistence/Models/Text.cs:9]: Why did you change from "label" to "text"? Just trying to follow the logic

**Review by petrochuk — COMMENTED** [review]
Maybe later but can we also add deserialization tests for actual yaml files?

File: src/Microsoft.PowerPlatform.PowerApps.Persistence/Models/Text.cs (hunk 1 above)
Comment (mizrael) [reply src/Microsoft.PowerPlatform.PowerApps.Persistence/Models/Text.cs:9]: renamed in favor of the actual control we have on the portal

File: src/Microsoft.PowerPlatform.PowerApps.Persistence/Yaml/ControlPropertiesCollectionConverter.cs [hunk 2]
```diff
@@ -41,12 +41,15 @@ public void WriteYaml(IEmitter emitter, object? value, Type type)
 
         emitter.Emit(new MappingStart());
 
-        foreach (var kv in collection)
+        var sortedKeys = collection.Keys.OrderBy(x => x);
```
Comment (petrochuk) [inline src/Microsoft.PowerPlatform.PowerApps.Persistence/Yaml/ControlPropertiesCollectionConverter.cs:44]: Should we specify culture for sorting?

File: src/Microsoft.PowerPlatform.PowerApps.Persistence/Yaml/SerializerBuilderExtensions.cs [hunk 3]
```diff
@@ -10,6 +10,8 @@ namespace Microsoft.PowerPlatform.PowerApps.Persistence.Yaml;
 
 internal static class SerializerBuilderExtensions
 {
+    private const string ControlPropertyName = "Control";
```
Comment (petrochuk) [inline src/Microsoft.PowerPlatform.PowerApps.Persistence/Yaml/SerializerBuilderExtensions.cs:13]: Probably builder extension is not the best place for this constant

Comment (mizrael) [reply src/Microsoft.PowerPlatform.PowerApps.Persistence/Yaml/SerializerBuilderExtensions.cs:13]: agreed. But we're not referencing it anywhere else, and we use "Control" only when serializing into YAML for now. I'd move it outside when we need to

File: src/Microsoft.PowerPlatform.PowerApps.Persistence/Yaml/ControlPropertiesCollectionConverter.cs (hunk 2 above)
Comment (mizrael) [reply src/Microsoft.PowerPlatform.PowerApps.Persistence/Yaml/ControlPropertiesCollectionConverter.cs:44]: good catch. `StringComparer.CurrentCultureIgnoreCase` ?

Comment (tehcrashxor) [reply src/Microsoft.PowerPlatform.PowerApps.Persistence/Yaml/ControlPropertiesCollectionConverter.cs:44]: Yeah, probably an Ordinal comparison.
I'm not sure what DocumentServer is doing for culture settings, but there could very well be a per-thread culture in response to requests.

**Review by petrochuk — APPROVED** [review]
